### PR TITLE
For #26286 improve firstRunScreenTest UI test coverage

### PR DESCRIPTION
**Pull request by AndiAJ**
For #26286 improve `firstRunScreenTest` UI test coverage ✅ successfully passed 100x on Firebase

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26286

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 1]
```diff
@@ -920,23 +1000,78 @@ private fun assertStartBrowsingButton() {
...
+    )
+
+private fun assertToolbarPlacementTopTitle() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_top_title"),
+            ).waitForExists(waitingTime),
+    )
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt:1074]: I'm starting to worry of the size of these robots and all these single use functions. Can we loop through a single function (e.g, verifyToolbarUI) that passes in the resource? (A pattern to start using in other places too)

Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt:1074]: Sounds good @AaronMT , did the same for clicking and long clicking different web page items.
I'll go ahead and refactor where applicable.

Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: Made the refactoring, please share your thoughts @AaronMT and @sv-ohorvath.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 2]
```diff
@@ -1114,6 +997,126 @@ private fun sponsoredShortcut(sponsoredShortcutTitle: String) =
...
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
+
+private fun assertAppItemsWithResourceIdAndText(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt:1043]: Looks awesome. I wonder if these should be moved to a MatcherHelper class or something of this sort, to be re-used in all robots. @AaronMT ?

**Review by sv-ohorvath — APPROVED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt (hunk 2 above)
Comment (AaronMT) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt:1043]: I'm OK with doing this outside this ticket, file a Jira.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**